File: examples/metal/vrf/example_2/go/main.go
package main

import (
	"github.com/equinix/pulumi-equinix/sdk/go/equinix/metal"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		example, err := metal.NewReservedIpBlock(ctx, "example", &metal.ReservedIpBlockArgs{
			Description: pulumi.String("Reserved IP block (192.168.100.0/29) taken from on of the ranges in the VRF's pool of address space."),
			ProjectId:   pulumi.Any(exampleEquinixMetalProject.Id),
			Metro:       pulumi.Any(exampleEquinixMetalVrf.Metro),
			Type:        pulumi.String("vrf"),
			VrfId:       pulumi.Any(exampleEquinixMetalVrf.Id),
			Cidr:        pulumi.Int(29),
			Network:     pulumi.String("192.168.100.0"),
		})
		if err != nil {
			return err
		}
		exampleVlan, err := metal.NewVlan(ctx, "exampleVlan", &metal.VlanArgs{
			Description: pulumi.String("A VLAN for Layer2 and Hybrid Metal devices"),
			Metro:       pulumi.Any(exampleEquinixMetalVrf.Metro),
			ProjectId:   pulumi.Any(exampleEquinixMetalProject.Id),
		})
		if err != nil {
			return err
		}
		_, err = metal.NewGateway(ctx, "exampleGateway", &metal.GatewayArgs{
			ProjectId:       pulumi.Any(exampleEquinixMetalProject.Id),
			VlanId:          exampleVlan.ID(),
			IpReservationId: example.ID(),
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: examples/metal/vrf/example_2/python/__main__.py
import pulumi
import pulumi_equinix as equinix

example = equinix.metal.ReservedIpBlock("example",
    description="Reserved IP block (192.168.100.0/29) taken from on of the ranges in the VRF's pool of address space.",
    project_id=example_equinix_metal_project["id"],
    metro=example_equinix_metal_vrf["metro"],
    type="vrf",
    vrf_id=example_equinix_metal_vrf["id"],
    cidr=29,
    network="192.168.100.0")
example_vlan = equinix.metal.Vlan("exampleVlan",
    description="A VLAN for Layer2 and Hybrid Metal devices",
    metro=example_equinix_metal_vrf["metro"],
    project_id=example_equinix_metal_project["id"])
example_gateway = equinix.metal.Gateway("exampleGateway",
    project_id=example_equinix_metal_project["id"],
    vlan_id=example_vlan.id,
    ip_reservation_id=example.id)
